### PR TITLE
Core: Add item.filler helper

### PR DESCRIPTION
**Pull request by Exempt-Medic**
## What is this fixing or adding?

Adds `item.filler` to go with `item.advancement`, `item.useful`, and `item.trap`

## How was this tested?

Tested by modifying Metroid Prime's code to use it

### CODE REVIEW

File: BaseClasses.py [hunk 1]
```diff
@@ -1264,6 +1264,10 @@ def useful(self) -> bool:
     def trap(self) -> bool:
         return ItemClassification.trap in self.classification
 
+    @property
+    def filler(self) -> bool:
+        return not self.classification
```
Comment (hesto2) [inline BaseClasses.py:1269]: 🙏

**Review by hesto2 — APPROVED** [review]

**Comment by NewSoupVi** [comment]
I think this should be `not (0b111 & self.classification)` to leave the door open for filler_skip_balancing and to work with worlds that us custom flags

**Comment by Exempt-Medic** [comment]
> I think this should be `not (0b111 & self.classification)` to leave the door open for filler_skip_balancing and to work with worlds that use their own custom flags

So the idea is that world-added flags can't change the "filler" status because filler is defined as a lack of trap/useful/progression? Thoughts on just being explicit with `not (self.trap or self.useful or self.advancement)` ?

**Comment by NewSoupVi** [comment]
> > I think this should be `not (0b111 & self.classification)` to leave the door open for filler_skip_balancing and to work with worlds that use their own custom flags
> 
> So the idea is that world-added flags can't change the "filler" status because filler is defined as a lack of trap/useful/progression? Thoughts on just being explicit with `not (self.trap or self.useful or self.advancement)` ?

I think I would say so. With the exception of trap ( :/ ), flags determine how core handles the item, and there is no way for a world to affect that with custom bits on the bitflag

**Review by hesto2 — APPROVED** [review]
Tested this in my apworld by pulling it in and verifying I could use it in my logic as expected (tests setup around this continued to pass, I verified they were not false positives)
```
        def is_hintable_filler_item(item: Item) -> bool:
            return item.filler and CivVIHintClassification.FILLER.value in self.options.pre_hint_items.value         
```

**Review by nicholassaylor — APPROVED** [review]
LGTM, this is what I'd expect such a helper to do

**Review by NewSoupVi — APPROVED** [review]
Marked for merge after hotfix window

~~How am I actually going to remember which PRs I've "marked for merge"? I didn't think this through~~

**Comment by Exempt-Medic** [comment]
> Marked for merge after hotfix window
> 
> ~How am I actually going to remember which PRs I've "marked for merge"? I didn't think this through~



> Marked for merge after hotfix window
> 
> ~How am I actually going to remember which PRs I've "marked for merge"? I didn't think this through~

Add waiting on other tag

**Pull request merged**